crates/wash-lib/src/app.rs

```diff
@@ -419,6 +435,55 @@ pub async fn load_app_manifest(source: AppManifestSource) -> anyhow::Result<AppM
         .context("app manifest loader timed out")?
 }
 
+/// Validate the contents of a manifest file and optionally validate the OCI references
+pub async fn validate_manifest_file(
+    manifest_contents: &str,
+    oci_check: bool,
+) -> Result<(Manifest, Vec<ValidationFailure>)> {
+    let manifest = serde_yaml::from_slice(manifest_contents.as_ref())
+        .context("failed to parse manifest content")?;
+
+    let failures = validate_manifest(&manifest).await?;
+
+    if oci_check {
+        let mut fetch_failures = Vec::new();
+        let image_references = extract_image_references(&manifest);
+        let fetcher = Fetcher::default();
+
+        for image in image_references {
+            if let Err(err) = fetcher.fetch_component(&image).await {
+                let fetch_failure = FetchFailure {
+                    image: image.clone(),
+                    error_message: err.to_string(),
+                };
+                // Collect the fetch failure
+                fetch_failures.push(fetch_failure);
+            }
+        }
+
+        // Display all fetch failures
+        for failure in &fetch_failures {
+            println!("{}", failure);
+        }
```

Comment:
NIT: This could be split out into a `validate_oci_references(refs: Vec<String>)`, what do you think?

Also, we probably want to use `eprintln` here to avoid messing with output to STDOUT